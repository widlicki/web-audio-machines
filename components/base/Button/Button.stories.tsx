import { ComponentMeta, ComponentStory } from '@storybook/react'
import Button from './Button'

export default {
  title: 'components/base/Button',
  component: Button,
} as ComponentMeta<typeof Button>

const Template: ComponentStory<typeof Button> = (args) => (
  <Button {...args}>Button</Button>
)

export const DefaultButton = Template.bind({})

export const DisabledButton = Template.bind({})
DisabledButton.args = {
  disabled: true,
}
