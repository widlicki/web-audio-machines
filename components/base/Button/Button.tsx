import { Button } from '@mui/material'
import theme from '../../../styles/theme'

const AppButton = ({ children, ...props }) => (
  <Button
    sx={{
      borderRadius: theme.border.radius.sm,
      padding: `${theme.spacing(2)} ${theme.spacing(4)} ${theme.spacing(2)}`,
      border: `${theme.border.size.sm} solid ${theme.border.color.light}`,
      backgroundColor: theme.palette.primary.light,
      color: 'common.white',
      '&:hover': {
        backgroundColor: theme.palette.primary.dark,
        cursor: 'pointer',
      },
      '&:disabled': {
        cursor: 'not-allowed',
        backgroundColor: theme.disabled.bgColor,
        border: `${theme.border.size.sm} solid ${theme.disabled.color}`,
        color: theme.disabled.color,
      },
    }}
    {...props}
  >
    {children}
  </Button>
)

export default AppButton
