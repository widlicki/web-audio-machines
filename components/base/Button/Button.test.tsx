import { fireEvent, render, screen } from '@testing-library/react'
import AppButton from './Button'
import { DefaultButton, DisabledButton, ToggleButton } from './Button.stories'

describe('Button Stories', () => {
  new Map([
    ['Default button', DefaultButton],
    ['Disabled button', DisabledButton],
    ['Toggle button', ToggleButton],
  ]).forEach((key) => {
    it(`renders ${key}`, () => {
      const component = <AppButton text="test" />
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})

describe('Button', () => {
  it('Should call onClick method when clicked', () => {
    const handleClick = jest.fn()

    const component = <AppButton onClick={handleClick} text="button" />
    render(component)
    fireEvent.click(screen.getByText(/button/i))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
