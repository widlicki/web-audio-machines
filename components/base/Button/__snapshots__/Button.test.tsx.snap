// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Stories renders function () { [native code] } 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-deftno-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    test
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button Stories renders function () { [native code] } 2`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-deftno-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    test
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
